test/test_fips.rb

```diff
@@ -8,4 +8,12 @@ def test_fips_mode_is_reentrant
     OpenSSL.fips_mode = false
   end
 
+  def test_fips_mode_returns_right_boolean
+    if OpenSSL::OPENSSL_FIPS
+      OpenSSL.fips_mode = true
+      OpenSSL.fips_mode == true
+      OpenSSL.fips_mode = false
+      OpenSSL.fips_mode == false
+    end
```

Comment:
I'll update, but when can the mode not be changed at runtime? Tested locally with fips enabled by default.